packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -487,11 +540,40 @@ export function makeVirtualObjectManager(
       if (initializing) {
         cache.remember(innerSelf);
       }
-      const representative = Far(tag, actualize(wrappedData));
+      const self = actualize(wrappedData);
+      let representative;
+      let exposedRepresentative;
+      const facetiousness = assessFacetiousness(self);
+      switch (facetiousness) {
+        case 'one': {
+          representative = Far(tag, self);
+          vrm.checkOrAcquireFacetNames(kindID, null);
+          exposedRepresentative = representative;
+          break;
+        }
+        case 'many': {
+          exposedRepresentative = {};
+          representative = [];
+          const facetNames = Object.getOwnPropertyNames(self).sort();
+          vrm.checkOrAcquireFacetNames(kindID, facetNames);
+          for (const facetName of facetNames) {
+            const facet = Far(`${tag} ${facetName}`, self[facetName]);
+            exposedRepresentative[facetName] = facet;
+            representative.push(facet);
+          }
+          harden(exposedRepresentative);
+          break;
+        }
+        case 'not':
+          assert.fail(`invalid self actualization for ${tag}`);
+        default:
+          assert.fail(`this can't happen`);
```

Comment:
```suggestion
          assert.fail(X`internal: unexpected facetiousness: ${q(facetiousness)}`);
```
We do need better handling in general for cannot-happen internal errors. In liveslots, it should kill the vat. In the kernel, it should panic. However we haven't made that convenient enough to start practicing, so no change suggested. For now, I sometimes remember to begin the error message with "internal: " to assure the user it isn't their fault and to encourage them to report it.